catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceDownloadManager.java

```diff
@@ -49,15 +55,16 @@
             LoggerFactory.getLogger(ReliableResourceDownloadManager.class);
 
     private ReliableResourceDownloaderConfig downloaderConfig;
+
     private DownloadStatusInfo downloadStatusInfo;
-    private ExecutorService executor;
+
+    private ListeningExecutorService executor;
 
     /**
-     * @param downloaderConfig
-     *            reference to the {@link ReliableResourceDownloaderConfig}
+     * @param downloaderConfig reference to the {@link ReliableResourceDownloaderConfig}
      */
     public ReliableResourceDownloadManager(ReliableResourceDownloaderConfig downloaderConfig,
-            DownloadStatusInfo downloadStatusInfo, ExecutorService executor) {
+            DownloadStatusInfo downloadStatusInfo, ListeningExecutorService executor) {
```

Comment:
One trick with the ExecutorService: if you want to provide a default implementation on an overloaded constructor, you can use a Java 8 `Supplier<ExecutorService>` to construct the Executor for you. This way, if the Executor is ever closed (which renders the object permanently not usable anymore), you can defer to the provided Supplier to create a new one.